src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -299,38 +309,62 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 				this.defaultRecommendedExtensionsContextKey.set(!this.configurationService.getValue<boolean>(ShowRecommendationsOnlyOnDemandKey));
 			}
 		}, this, this.disposables);
+
+		modes.SuggestRegistry.register({ scheme: 'extensions', pattern: '**/searchinput', hasAccessToAllModels: true }, {
+			triggerCharacters: ['@'],
+			provideCompletionItems: (model: ITextModel, position: Position, _context: modes.SuggestContext) => {
+				return {
+					suggestions: this.autoComplete(model.getValue(), position.column).map(item => (
+						{
+							label: item.fullText,
+							insertText: item.fullText,
+							overwriteBefore: item.overwrite,
+							sortText: item.fullText.indexOf(':') === -1 ? 'a' : item.fullText.indexOf('sort') !== -1 ? 'b' : 'c', // things with no colon; sorts; everything else
```

Comment:
Is the idea for this sorting to show the pre-defined views first, followed by the most commonly used "@sort", and then the rest?
This makes `tag` and `ext` go to the very bottom. We can probably bring them up.

![image](https://user-images.githubusercontent.com/16890566/42855982-fd396764-89f7-11e8-8be2-810b81e26729.png)

Reply:
It doesn't make sense to me to have "sort:" and "category:" in the list, as a user would never want to select them. We could just change the sort algorithm to put "ext:" and "tag:" before "category:*"?

Reply:
sure, that works